docs/config/node.rst

```diff
@@ -851,15 +851,19 @@ Queries
   | *Default:* ``8192``
   | *Runtime:* ``no``
 
-  This setting defines the maximum number of elements an array can have so that
-  the ``!= ANY()``, ``LIKE ANY()``, ``ILIKE ANY()``, ``NOT LIKE ANY()`` and the
-  ``NOT ILIKE ANY()`` :ref:`operators <gloss-operator>` can be applied on it.
+  This setting limits the number of boolean clauses that can be generated by
+  ``!= ANY()``, ``LIKE ANY()``, ``ILIKE ANY()``, ``NOT LIKE ANY()`` and
+  ``NOT ILIKE ANY()`` :ref:`operators <gloss-operator>` on arrays in order to
+  prevent executing queries that may result in heavy memory consumptions
+  causing nodes to crash with ``OutOfMemory`` exceptions. Throws
+  ``TooManyClauses`` exceptions when exceeded.
```

Comment:
```suggestion
  ``TooManyClauses`` errors when the limit is exceeded.
```
From a quick look we don't use `exceptions` in docs as this is Java specific and we prefer the more generic: `errors`.